packages/pwa-kit-runtime/src/utils/ssr-config.server.js

```diff
@@ -8,59 +8,82 @@
+ *      E.g., `my-config/production.json`, or `my-config/development.json`
  *
- * local.ext - Only loaded on local development environments, this file is used if you want a custom
- * configuration that will not be used on deployed remote environments.
+ * `local.{ext}` - Only loaded on local development environments
+ *  - This file is used if you want a configuration that will not be used on deployed remote environments.
+ *  - Custom config file named with the `local` suffix. E.g., `my-config.local.json`
+ *  - If the custom config is a directory, default naming conventions will be apply
+ *      E.g., `my-config/local.json`
  *
- * default.ext - If you have no requirement for environment specific configurations the `default`
- * config file will be used.
+ * `default.{ext}`
+ *  - If you have no requirement for environment specific configurations the `default` config file
+ *      will be used.
+ *  - Custom config file uses its own name. E.g., `my-config.json`
+ *  - If the custom config file is a directory, default naming conventions will be apply
+ *      E.g., `my-config/default.json`
  *
- * package.json - If none of the files after have been found the `mobify` object defined in the
- * projects `package.json` file.
+ * `package.json`
+ *  - If none of the files after have been found the `mobify` object defined in the projects
+ *      `package.json` file.
+ *  - This file only used for default application configuration.
  *
  * Each file marked with `ext` can optionally be terminated with `js`, `yml|yaml` or
  * `json`. The file loaded is also determined based on that precidence of file extension.
  *
- * @param {Object} opts.buildDirectory - Option path to the folder containing the configution. Byt default
- * it is the `build` folder when running remotely and the project folder when developing locally.
- * @returns - the application configuration object.
+ * @param {object} opts
+ * @param {string} [opts.buildDirectory] - Option path to the folder containing the configution.
+ *      By default it is the `build` folder when running remotely and the project folder when developing locally.
+ * @param {string} [opts.configFile] - Custom config file/path that containing the configution
+ *
+ * @returns {object} - the application configuration object.
  */
 /* istanbul ignore next */
 export const getConfig = (opts = {}) => {
-    const {buildDirectory} = opts
+    const {buildDirectory, configFile = 'config'} = opts
     const configDirBase = IS_REMOTE ? 'build' : ''
-    let targetName = process?.env?.DEPLOY_TARGET || ''
-
-    const targetSearchPlaces = SUPPORTED_FILE_TYPES.map((ext) => `config/${targetName}.${ext}`)
-    const localeSearchPlaces = SUPPORTED_FILE_TYPES.map((ext) => `config/local.${ext}`)
-    const defaultSearchPlaces = SUPPORTED_FILE_TYPES.map((ext) => `config/default.${ext}`)
-
     const searchFrom = buildDirectory || process.cwd() + '/' + configDirBase
+    const isCustom = configFile !== 'config'
```

Comment:
Do any of these modifications change the default behaviour of the `getConfig` function? If so, this would be a breaking change and we would need to release a new major version.

Reply:
This feature don't change any existing behaviour of the `getConfig`.